src/master-playlist-controller.js

```diff
@@ -143,14 +149,23 @@ export default class MasterPlaylistController extends videojs.EventTarget {
 
     this.masterPlaylistLoader_.on('mediachange', () => {
       let media = this.masterPlaylistLoader_.media();
+      let requestTimeout = (this.masterPlaylistLoader_.targetDuration * 1.5) * 1000;
 
       this.mainSegmentLoader_.abort();
 
+      // If we don't have any more available playlists, we don't want to
+      // timeout the request.
+      if (this.masterPlaylistLoader_.onLowestEnabledRendition_()) {
```

Comment:
This is a total nitpick but `on*` functions names are historically used to denote an event handler. Something like `isLowestEnabledRendition_` would be potentially less confusing.